cds_migrator_kit/videos/weblecture_migration/transform/xml_processing/rules/base.py

```
@@ -0,0 +1,105 @@
+    StringValue,
+    clean_str,
+    clean_val,
+)
+
+from ...models.base import model
+from ..quality.contributors import get_contributor_role
+
+
+@model.over("legacy_recid", "^001")
+def recid(self, key, value):
+    """Record Identifier."""
+    self["recid"] = value  # TODO recid??
+    return int(value)
+
+
+@model.over("title", "^245__")
+def title(self, key, value):
+    """Translates title."""
+    title = StringValue(value.get("a"))
+    if value.get("b"):
+        title = StringValue(value.get("a") + " : " + value.get("b"))
+    title.required()
+    return {"title": title.parse()}
+
+
+@model.over("description", "^520__")
+def description(self, key, value):
+    """Translates description."""
+    description_text = StringValue(value.get("a")).parse()
+
+    return description_text
+
+
+@model.over("language", "^041__")
+@require(["a"])
+@strip_output
+def language(self, key, value):
+    """Translates language field."""
+    lang = clean_str(value.get("a"))
+    if lang:
+        lang = lang.lower()
+    try:
+        return pycountry.languages.lookup(lang).alpha_2.lower()
+    except (KeyError, AttributeError, LookupError):
+        raise UnexpectedValue(field=key, subfield="a")
+
+
+@model.over("contributors", "^100__")
+@for_each_value
+@require(["a"])
+def creators(self, key, value):
+    """Translates the creators field."""
+    role = value.get("e")
+    if role:
+        role = get_contributor_role("e", role)
+    beard = value.get("9")
+    if beard is not None and beard != "#BEARD#":
+        # checking if anything else stored in this field
+        # historically it was some kind of automatic script tagging
+        # and it should be ignored if value == #BEARD#
+        raise UnexpectedValue(field=key, subfield="9", value=beard)
+    name = value.get("a").strip()
+    contributor = {"name": name}
+    if role:
+        contributor.update({"role": role})
+    # TODO contributor affiliation will be implemented
```

can we add a check to `raise UnexpectedValue(field=key, subfield="u", value=...)` is we find an affiliation? So, we are sure when that happens?

---

Decided to migrated them as array of strings